srv/components/provisioner/scripts/cortx_prep.sh

```diff
@@ -0,0 +1,291 @@
+                fi
+                repo_url=$2
+                shift 2
+                ;;
+            -h|--help)
+                usage; exit 0
+            *) echo "Invalid option $1"; usage; exit 1;;
+        esac
+    done
+}
+
+config_local_salt()
+{
+
+    # 1. Point minion to local host master in /etc/salt/minion
+    # 2. Restart salt-minion service
+    # 3. Check if 'salt srvnode-0 test.ping' works
+
+    if ! rpm -qa | grep -iEwq "salt|salt-master|salt-minion|cortx-prvsnr"; then
+        echo "ERROR: salt packages are not installed, please install salt packages and try again." | tee -a ${LOG_FILE}
+        exit 1
+    fi
+
+    minion_file="${PRVSNR_ROOT}/srv/components/provisioner/salt_minion/files/minion_factory"
+    master_file="${PRVSNR_ROOT}/srv/components/provisioner/salt_master/files/master"
+
+    yes | cp -f "${master_file}" /etc/salt/master
+    yes | cp -f "${minion_file}" /etc/salt/minion
+
+    #local line_to_replace=$(grep -m1 -noP "master: " /etc/salt/minion|tail -1|cut -d: -f1)
+    #echo "Setting minion to local master in /etc/salt/minon" >> ${LOG_FILE}
+    #yes | cp -f /etc/salt/minion /etc/salt/minion.bkp
+    #sed -i "${line_to_replace}s|^master:.*|master: ${localhost_ip}|" /etc/salt/minion
+    echo "Done" | tee -a ${LOG_FILE}
+
+    echo $minion_id > /etc/salt/minion_id
+
+    echo "Restarting the required services" | tee -a ${LOG_FILE}
+    systemctl start salt-master
+    systemctl restart salt-minion
+    echo "Done" | tee -a ${LOG_FILE}
+
+    echo "INFO: Waiting for key of $minion_id to become connected to salt-master" >> ${LOG_FILE}
+    try=1; max_tries=10
+    until salt-key --list-all | grep "srvnode-0" >/dev/null 2>&1
+    do
+        if [[ "$try" -gt "$max_tries" ]]; then
+            echo "ERROR: Key for salt-minion $minion_id not listed after $max_tries attemps." >> ${LOG_FILE}
+            echo "ERROR: Provisioner configuration manager failed for key acceptance" | tee -a ${LOG_FILE}
+            salt-key --list-all >&2
+            exit 1
+        fi
+        try=$(( $try + 1 ))
+        sleep 5
+    done
+    echo "Key for $minion_id is listed." >> ${LOG_FILE}
+    echo "Accepting the salt key for minion $minion_id" >> ${LOG_FILE}
+    salt-key -y -a $minion_id
+
+    # Check if salt '*' test.ping works
+    echo "Waiting for Provisioner configuratoin manager to become ready" | tee -a ${LOG_FILE}
+    try=1; max_tries=10
+    until salt -t 1 $minion_id test.ping >/dev/null 2>&1
+    do
+        if [[ "$try" -gt "$max_tries" ]]; then
+            echo "ERROR: minion srvnode-0 seems still not ready after $max_tries attempts." >> ${LOG_FILE}
+            echo "ERROR: Provisioner configuration manager failed to connect to local server" | tee -a ${LOG_FILE}
+            exit 1
+        fi
+        try=$(( $try + 1 ))
+    done
+    echo "Minion on Server A started successfully" >> ${LOG_FILE}
+    echo "Done" | tee -a ${LOG_FILE}
+}
+
+create_factory_user()
```

Comment:
Remove this function, as we plan to create users as part of node finalize step towards the end of manufacturing...

Reply:
removed